app-editors/vscode/vscode-1.56.0.ebuild

```diff
@@ -0,0 +1,74 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop pax-utils xdg
+
+DESCRIPTION="Multiplatform Visual Studio Code from Microsoft"
+HOMEPAGE="https://code.visualstudio.com"
+SRC_URI="
+	amd64? ( https://update.code.visualstudio.com/${PV}/linux-x64/stable -> ${P}-amd64.tar.gz )
+	arm? ( https://update.code.visualstudio.com/${PV}/linux-armhf/stable -> ${P}-arm.tar.gz )
+	arm64? ( https://update.code.visualstudio.com/${PV}/linux-arm64/stable -> ${P}-arm64.tar.gz )
+"
+
+RESTRICT="mirror strip bindist"
+
+LICENSE="Microsoft-VSCode"
```

Comment:
Is it a standard approach to list the licences of every library used to make the app work ? Or are you talking about the "extensions" ? vscode works like an app store and the user accepts the licence by installing them no ?

Reply:
You wouldn't list licenses of dependencies (installed by portage), but in this case it's bundled and is installed by "this" package and so has to be listed. Microsoft's license doesn't replace other projects licenses and they all apply.

Reply:
Okay I'll investigate this further. Thanks!

Reply:
Okay I added the title of all the licenses of all the libraries and binaries used. It seems that a big number of these licenses are not available in Gentoo. Some license names in Gentoo are versioned with a `.0`, like `ECL-2.0` in their name and some don't, like `BSD-2`. It is extremely annoying since sometimes they are not detected because of the missing/extra `.0` in the license name, `repoman` is complaining that I split the licenses to one per line with leading spaces, I think it is better that way, I don't know what you think about it. I suppose I must install every single lacking license right ? And know under which category they go, I want to cry... :cry:

Reply:
I started adding a few licenses, but I think it's better to write an  HTML parser in python that downloads licenses from this webpage https://spdx.org/licenses/  and saves them into a file. I will write that script, then add it to GURU and tell the others about it so they can use it for their packages.

Reply:
I opened a bug to get feedback on all the licenses that need to be added https://bugs.gentoo.org/789555